atom/browser/extensions/tab_helper.cc

```diff
@@ -253,6 +253,9 @@ void TabHelper::WillCloseWindow(bool* prevent_default) {
     browser()->window()->Deactivate();
     browser()->window()->Hide();
   }
+  if (!browser_shutdown::IsTryingToQuit()) {
```

Comment:
this can be handled by calling prevent default `app.on('close'` in browser-laptop